src/Microsoft.FeatureManagement.AspNetCore/TargetingHttpContextMiddleware.cs

```diff
@@ -0,0 +1,54 @@
+﻿// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT license.
+//
+
+using Microsoft.AspNetCore.Http;
+using Microsoft.FeatureManagement.FeatureFilters;
+using System;
+using System.Threading.Tasks;
+
+namespace EvaluationDataToApplicationInsights.Telemetry
+{
+    /// <summary>
+    /// Used to add targeting information to http context. This allows synronous code to access targeting information.
+    /// </summary>
+    public class TargetingHttpContextMiddleware
+    {
+        private readonly RequestDelegate _next;
+
+        /// <summary>
+        /// Creates an instance of the TargetingHttpContextMiddleware
+        /// </summary>
+        public TargetingHttpContextMiddleware(RequestDelegate next) { 
```

Comment:
I suggest we add a logger here.
```suggestion
        public TargetingHttpContextMiddleware(RequestDelegate next, , ILoggerFactory loggerFactory) { 
```
When targetingContext == null, we can log something for debug.